includes/subscriptions/class-wc-payments-subscriptions-event-handler.php

```diff
@@ -244,8 +264,36 @@ public function handle_invoice_payment_failed( array $body ) {
 			}
 		}
 
-		// Translators: %d Number of failed renewal attempts.
-		$subscription->add_order_note( sprintf( _n( 'WCPay subscription renewal attempt %d failed.', 'WCPay subscription renewal attempt %d failed.', $attempts, 'woocommerce-payments' ), $attempts ) );
+		if ( $error_details ) {
+			$subscription->add_order_note(
+				sprintf(
+					// Translators: %1$d Number of failed renewal attempts. %2$s contains failure message, %3$s contains error code.
+					_n(
+						'WCPay subscription renewal attempt %1$d failed with the following message "%2$s" and failure code <code>%3$s</code>',
+						'WCPay subscription renewal attempt %1$d failed with the following message "%2$s" and failure code <code>%3$s</code>',
+						$attempts,
+						'woocommerce-payments'
+					),
+					$attempts,
+					$error_details,
+					$error_code
+				)
+			);
+			$order->add_order_note(
+				sprintf(
+					// Translators: %1$s contains failure message. %2$s contains error code.
+					__(
+						'Payment for the order failed with the following message: "%1$s" and failure code <code>%2$s</code>',
+						'woocommerce-payments'
+					),
+					$error_details,
+					$error_code
+				)
+			);
+		} else {
+			// Translators: %d Number of failed renewal attempts.
+			$subscription->add_order_note( sprintf( _n( 'WCPay subscription renewal attempt %d failed.', 'WCPay subscription renewal attempt %d failed.', $attempts, 'woocommerce-payments' ), $attempts ) );
```

Comment:
Would it make sense to refer to the plugin's name, here? Or are there other instances where we refer to `WCPay` in the notes?
```suggestion
						'WooPayments subscription renewal attempt %1$d failed with the following message "%2$s" and failure code <code>%3$s</code>',
						'WooPayments subscription renewal attempt %1$d failed with the following message "%2$s" and failure code <code>%3$s</code>',
						$attempts,
						'woocommerce-payments'
					),
					$attempts,
					$error_details,
					$error_code
				)
			);
			$order->add_order_note(
				sprintf(
					// Translators: %1$s contains failure message. %2$s contains error code.
					__(
						'Payment for the order failed with the following message: "%1$s" and failure code <code>%2$s</code>',
						'woocommerce-payments'
					),
					$error_details,
					$error_code
				)
			);
		} else {
			// Translators: %d Number of failed renewal attempts.
			$subscription->add_order_note( sprintf( _n( 'WooPayments subscription renewal attempt %d failed.', 'WooPayments subscription renewal attempt %d failed.', $attempts, 'woocommerce-payments' ), $attempts ) );
```

Reply:
I wanted to change the name, but `grep` found a few other usages across the plugin codebase, so I opted to make it a separate change, instead of extending scope of this pull request.

Reply:
Ah, I see - there's also occurrences like:
```php
$subscription->add_order_note( __( 'Suspended WCPay Subscription because subscription status changed to on-hold.', 'woocommerce-payments' ) );
$subscription->add_order_note( __( 'The subscription\'s next payment date has been updated to match WCPay server.', 'woocommerce-payments' ) );
$subscription->add_order_note( __( 'Suspended WCPay Subscription in invoice.upcoming webhook handler because subscription next_payment date is 0.', 'woocommerce-payments' ) );
```

All good 👍